qiskit_nature/algorithms/initial_points/mp2_initial_point.py

```diff
@@ -0,0 +1,363 @@
+
+    If the Hartree-Fock reference energy was found in
+    :class:`~qiskit_nature.properties.second_quantization.electronic.ElectronicEnergy`, it will
+    be used to compute the absolute MP2 energy via :meth:`~get_energy`.
+    """
+
+    def __init__(self, threshold: float = 1e-12) -> None:
+        super().__init__()
+        self.threshold: float = threshold
+
+        self._grouped_property: GroupedSecondQuantizedProperty | None = None
+        self._integral_matrix: np.ndarray | None = None
+        self._orbital_energies: np.ndarray | None = None
+        self._reference_energy: float = 0.0
+
+        self._ansatz: UCC | None = None
+        self._excitation_list: list[tuple[tuple[int, ...], tuple[int, ...]]] | None = None
+
+        self._corrections: dict[str, tuple[float, float]] | None = None
+        self._missing_input_error_message: str = (
+            "Not enough information has been provided to compute the MP2 corrections. "
+            "Set `grouped_property` and `ansatz` or call `compute` with them as arguments."
+        )
+
+    @property
+    def grouped_property(self) -> GroupedSecondQuantizedProperty:
+        """The grouped property."""
+        return self._grouped_property
+
+    @grouped_property.setter
+    def grouped_property(self, grouped_property: GroupedSecondQuantizedProperty) -> None:
+        """The grouped property.
+
+        The grouped property is required to contain the ElectronicEnergy, which must contain
+        the two-body molecular orbital matrix and the orbital energies. Optionally, it will also
+        use the Hartree-Fock reference energy to compute the absolute energy.
+
+        Raises:
+            QiskitNatureError: If
+                :class:`~qiskit_nature.properties.second_quantization.electronic.ElectronicEnergy`
+                is missing or the two-body molecular orbital matrix or the orbital energies are not
+                found.
+        """
+
+        electronic_energy: ElectronicEnergy | None = grouped_property.get_property(ElectronicEnergy)
+        if not isinstance(electronic_energy, ElectronicEnergy):
+            raise QiskitNatureError(f"ElectronicEnergy not in grouped property: {grouped_property}")
+
+        two_body_mo_integral: ElectronicIntegrals | None = (
+            electronic_energy.get_electronic_integral(ElectronicBasis.MO, 2)
+        )
+        if not isinstance(two_body_mo_integral, ElectronicIntegrals):
+            raise QiskitNatureError(
+                f"Two body MO electronic integral not in grouped property: {grouped_property}"
+            )
+
+        orbital_energies: np.ndarray | None = electronic_energy.orbital_energies
+        if not isinstance(orbital_energies, np.ndarray):
+            raise QiskitNatureError(f"Orbital energies not in grouped property: {grouped_property}")
+
+        # Invalidate any previous computation.
+        self._corrections = None
+
+        self._integral_matrix = two_body_mo_integral.get_matrix()
```

Comment:
This implies that the current implementation will only work for cases where the alpha and beta spin MOs are identical (i.e. came out of a restricted spin calculation). While this is okay for the time being, you should check that this is indeed the case, and, if it is not, raise an error here:
```suggestion
        self._integral_matrix = two_body_mo_integral.get_matrix()
        if not np.allclose(self._integral_matrix, two_body_mo_integral.get_matrix(2)):
            raise NotImplementedError("MP2InitialPoint only supports restricted-spin setups.")
```
The error message can be improved slightly. You should also open an issue which tracks the progress of extending this implementation to unrestricted-spin cases (should not be too hard).